### DISCUSSION TITLE
Remove Annotated text and non-annotated text separately.

### DISCUSSION THREAD

**Discussion by siddhesh-provision** [pymupdf/PyMuPDF#3365] [Looking for help] [answered]
I want to separate out annotated and non-annotated text. Its easy to get annotated text, and all text, but not non-annotated text.

Here's the order of operations Ive tried:
1. Get annotated text
2. Delete annotations
3. Extract all text

But that doesnt work as expected since deleting annotations simply deletes the annotations and not the text. Can somebody show me how to do this? Or is there some way to delete annotations along with text?

Here's how I'm deleting annotations:

```
for a in page.annots():
     # get text
    page.delete_annot(a)
```

**Comment by JorjMcKie** [comment, marked as answer]
Your problem is a misconception:
There is no fixed connection between what an annotation rectangle happens to cover and the stuff underneath that rectangle.
An annotation can be defined anywhere: over images, vector graphics and also empty space.
Therefore the annotated text (this is a misleading term - as explained!) does not care or take note of whether it is covered by an annotation.

What you **_can_** do is removing the text underneath an annot rectangle by defining a redaction annotation with the original annot.rect and then **_applying_** all defined redactions. This will remove desired text.

**Reply by siddhesh-provision** [reply]
I see what you mean. Thank you for the response!
